sunkit_instruments/abstractions/channels.py

```diff
@@ -0,0 +1,29 @@
+"""This module defines abstractions related to instrument channels."""
+
+__all__ = ["AbstractChannel"]
+
+import abc
+
+import astropy.units as u
+
+
+class AbstractChannel(abc.ABC):
+    """An abstract base class for defining instrument channels.
+
+    .. caution::
+
+       This abstract class is still under development and may change
+       in the near future.
+    """
+
+    @abc.abstractmethod
+    def effective_area(self) -> u.cm**2:
+        ...
+
+    @abc.abstractmethod
+    def gain(self) -> u.DN / u.photon:
+        ...
```

Comment:
Do we want to have `gain` be a required part of the interface?

Reply:
Will there be cases where a gain is a bad idea or impossible to work out?

Reply:
Maybe. It could always just be set to 1?

Reply:
I am happy to leave this decision up to the people who work with observational data and have a good understanding of what gain actually is. 🙃  Defaulting to 1 sounds quite reasonable though.

Reply:
I think it should be left without any value such that anyone using the abstract class has to make the explicit assumption about what their gain is.